tensorboard/uploader/auth_test.py

```diff
@@ -149,5 +155,305 @@ def test_read_invalid_json_file(self):
             store.read_credentials()
 
 
+class FakeInstalledAppFlow:
+    """A minimal fake for the InstalledApp flow with the function we call.
+
+    This is a fake of a publicly available class that should already be tested,
+    so we mostly want to test high-level interactions with it.
+    """
+
+    def __init__(self, credentials=None, raiseError=False):
+        if not credentials and not raiseError:
+            ValueError("credentials cannot be None when raiseError is False")
+        if credentials and raiseError:
+            ValueError("credentials must be None when raiseError is True")
+        self.run_local_server_was_called = False
+        self._creds = credentials
+        self._raiseError = raiseError
+
+    def run_local_server(self, port=8080):
+        self.run_local_server_was_called = True
+        if self._raiseError:
+            raise webbrowser.Error()
+        return self._creds
+
+
+class AuthenticateUserTest(tb_test.TestCase):
+    def setUp(self):
+        super().setUp()
+        # Used to estimate if a browser is available in this env.
+        self.os_env_display_fn = self.enter_context(
+            mock.patch.object(os, "getenv")
+        )
+
+        self.mocked_installed_auth_flow_creator_fn = self.enter_context(
+            mock.patch.object(auth_flows.InstalledAppFlow, "from_client_config")
+        )
+
+        self.mocked_device_auth_flow = self.enter_context(
+            mock.patch.object(
+                auth, "_LimitedInputDeviceAuthFlow", autospec=True
+            )
+        )
+
+    def test_authenticate_user__no_force_console_override__has_display__uses_installed_app_flow(
+        self,
+    ):
+        self.os_env_display_fn.return_value = "some_display"
+
+        fake_auth_flow = FakeInstalledAppFlow(
+            credentials=Credentials("fake_access_token")
+        )
+        self.mocked_installed_auth_flow_creator_fn.return_value = fake_auth_flow
+
+        auth.authenticate_user()
+
+        self.mocked_installed_auth_flow_creator_fn.assert_called_once()
+        self.assertTrue(fake_auth_flow.run_local_server_was_called)
+        self.mocked_device_auth_flow.assert_not_called()
+
+    def test_authenticate_user__no_force_console_override__no_display__uses_device_flow(
+        self,
+    ):
+        self.os_env_display_fn.return_value = None
+
+        auth.authenticate_user()
+
+        self.mocked_installed_auth_flow_creator_fn.assert_not_called()
+        self.mocked_device_auth_flow.assert_called_once()
+        self.mocked_device_auth_flow.return_value.run.assert_called_once()
+
+    def test_authenticate_user__no_force_console_override__has_display__webbrowser_error__uses_device_flow(
```

Comment:
Perhaps just name "test_uses_device_flow_on_webbrowser_error(self):"

Reply:
Updated to something different, but it's a bit shorter and easier to read.